music21/romanText/tsvConverter.py

```diff
@@ -291,14 +510,26 @@ def importTsv(self):
         fileName = self.tsvFileName
 
         with open(fileName, 'r', encoding='utf-8') as f:
-            data = []
-            for row_num, line in enumerate(f):
-                if row_num == 0:  # Ignore first row (headers)
-                    continue
-                values = line.strip().split('\t')
-                data.append([v.strip('\"') for v in values])
+            tsvreader = csv.reader(f, delimiter='\t', quotechar='"')
+            # The first row is the header
+            self._get_heading_indices(next(tsvreader))
+            return list(tsvreader)
+
+    def _makeTabChord(self, row):
+        '''
+        Makes a TabChord out of a list imported from TSV data
+        (a row of the original tabular format -- see TsvHandler.importTsv()).
+        '''
+        # this method replaces the previously stand-alone makeTabChord function
+        thisEntry = self._tab_chord_cls()
+        for i, (name, type_) in self._head_indices.items():
+            setattr(thisEntry, name, type_(row[i]))
```

Comment:
Here we are setting the attributes of TabChord or TabChordV2 (e.g., `measure`, `chord`, etc.) according to the values in the tsv row.
I use `setattr` here because TabChord and TabChordV2 have different attributes (corresponding to the different columns of the DCML tsv versions). The `name` values will only be those from the appropriate HEADERS because of lines 500-503, so we shouldn't  be setting any unanticipated attributes. We can add another layer of checks here though (e.g., `if name not in self.heading_names: continue`). Or I am open to suggestions for other ways of completing the same task.

Reply:
If they have separate attributes, then you can call the same method name on the two different classes and they can work.  So move the logic to the TabChord/TabChordV2 itself.

Reply:
So if I understand correctly, within these methods, you would like us to assign explicitly to each attribute, e.g.,
```
if "chord" in head_indices:
    i, type_to_coerce_to = head_indices["chord"]
    self.chord = type_to_coerce_to(row[i])
```
Repeated for each column name.

I can do this, but it is a lot of typing and will need to be altered any time we add or subtract columns. What about something like
```
for col_name, (i, type_to_coerce_to) in head_indices.items():
    if col_name in self.headers:
        setattr(self, col_name, type_to_coerce_to(row[i]))
```